php/WC_Gateway_Blockonomics.php

```diff
@@ -114,16 +114,38 @@ public function handle_requests()
         include_once 'Blockonomics.php';
         $blockonomics = new Blockonomics;
 
+        $order_hash = "";
+
+        if ($show_order) {
+            $order_hash = $show_order;
+        } else if ($select_crypto) {
+            $order_hash = $select_crypto;
+        } else if ($finish_order) {
+            $order_hash = $finish_order;
+        } else if ($get_order) {
+            $order_hash = $get_order;
+        }
+        
+        $order_id = "";
+
+        if ($order_hash) {
+            $order_id = $blockonomics->decrypt_hash($order_hash);
+            if (empty(wc_get_order($order_id))) {
```

Comment:
The check for woocommerce order should be moved into decrypt_hash function to ensure this check always happens after decrypting the hash, such as in blockonomics_nojs_checkout.php.